android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java

```diff
@@ -190,21 +190,27 @@ public String getCarrier() {
   }
 
   @ReactMethod
-  public Integer getTotalDiskCapacity() {
+  public long getTotalDiskCapacity() {
     try {
-      StatFs root = new StatFs(Environment.getRootDirectory().getAbsolutePath());
-      return root.getBlockCount() * root.getBlockSize();
+      File sdcardDir = Environment.getExternalStorageDirectory();
+      StatFs statFs = new StatFs(sdcardDir.getPath());
+      long blockSize = statFs.getBlockSizeLong();
+      long totalSize = statFs.getBlockCountLong();
+      return blockSize * totalSize;
```

Comment:
These StatFS.getXxxLong() APIs are for Android API >= 18, but react-native still supports down to API16. The appropriate way to handle this is more like the approach here with an API guard:

https://github.com/rebeccahughes/react-native-device-info/pull/350/files#diff-d2dc05e092aa2dcf49063f6d1f012068R162

(though I think a long is better than a float as a return value, unlike that link...)